lib/XRay/Formats/Mf2.php

```diff
@@ -697,18 +705,22 @@ private static function parseAsHCard($item, $http, $url, $authorURL=false) {
       if($p == 'url' && $authorURL) {
         // If there is a matching author URL, use that one
         $found = false;
-        foreach($item['properties']['url'] as $url) {
-          if(self::isURL($url)) {
-            $url = \p3k\XRay\normalize_url($url);
-            if($url == \p3k\XRay\normalize_url($authorURL)) {
-              $data['url'] = $url;
-              $found = true;
+        if (array_key_exists('url', $item['properties']) and is_array($item['properties']['url'])) {
```

Comment:
One thing I'm trying to be consistent about in XRay is returning the same data structure regardless of the data found. So in this case, if there is no `url` property found, it should still return a `url` in the XRay response. I can't remember the normal behavior whether it should return an empty string or `false` or `null` but it should do whatever is done elsewhere.

Reply:
It turns out XRay does this already, albeit slightly inconsistently. The return value for `$data['data']` is filled at creation with `'url' => null`, and later on, if no URL property was found, it’s replaced with the value of the `$authorUrl` which was passed to `parseAsHCard`, which defaults to `false` but from my testing seems to be set to the page URL if the h-card was the main content found on the page.

So your code already makes sure that there’ll always be a `url` property, but depending on the situation it’ll either be `false` or a string, which is inconsistent with the other pre-filled properties (name and photo) which will be `null` if no matching property is found.

Reply:
So I don’t need to change the code I added, but and in this case I can test that the URL was set to the page URL, but in theory there’s a chance that you could end up with a weird h-card array like
```php
[
  'name' => null,
  'photo' => null,
  'url' => false
]
```

and I suppose whether or not that’s a problem is for another issue.

Reply:
oh! well cool, that works. annoying that it's false/null but that's a different problem.